internal-services/catalog/file-updates-process-file-updates-task.yaml

```diff
@@ -118,11 +125,44 @@ spec:
                   valueSize="$(yq "${key}" ${targetFile} |wc -l)"
                   startBlock=$(( ${foundAt} + ${blankLinesBeforeYaml} ))
 
+                  # the replacement should be a sed expression using "|" as separator
+                  if [[ $(tr -dc "|" <<< "${replacement}" | wc -m ) != 3 ]]; then
+                      updateError='Replace expression should be in `|search|replace|` format'
+                      break
+                  fi
+
                   # run the replace
                   sed -i "${startBlock},+${valueSize}s${replacement}" "${targetFile}"
+
+                  # get the replace part of "|search|replace|"
+                  replaceStr=$(awk -F"|" '{print $3}' <<< "${replacement}")
+
+                  # when the value is a text block we must make sure
+                  # only a single line was replaced and that the result
+                  # block has the same number of lines as before
+                  replacedBlockLines=$(sed -ne "${startBlock},+${valueSize}p" "${targetFile}" |wc -l)
+                  if [[ $replacedBlockLines != $(( $valueSize +1 )) ]]; then
+                      updateError="Text block size differs from the original"
```

Comment:
are there any details on what the differences might be? Can we show the user?

Reply:
Maybe the error string could be better. The idea here is: In  case the replace expression was miswritten as for instance `|.*|image: some_image|`, if the value is a block like this:
```
   image: image_a
   image: image_b
   image: image_c
   image: image_d
```
The result would be:
```
  image: some_image
```
The idea is prevent cases like this. If the user wants multiple replacements in the same block the user needs to define different replacements for each line.

Maybe I should add something like: `Too many lines replaced. Check if the expression is not too greedy" and print the expression.

Reply:
yeah that works!